src/IndexService/Config/AbstractConfig.php

```diff
@@ -173,15 +173,10 @@ public function getTenantName(): string
     /**
      * Returns configured attributes for product index
      *
-     * @return Attribute[]
+     * @return Attribute[]|null
      */
-    public function getAttributes(): array
+    public function getAttributes(): ?array
```

Comment:
also remove legacy code from `getSearchAttributes`

Reply:
I initially only removed the legacy code here where the exception is thrown. Then I tried to install it again and got the following error message, hence why I made it nullable. 
In AbstractConfig.php line 180:
                                                                                                                                                  
  Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Config\AbstractConfig::getAttributes(): Return value must be of type array, null returned  
                                                                 
Guess it has something to do with that it is declared nullable. We could also change that to an array instead of null? Something like this?
 /**
     * @var Attribute[]
     */
    protected array $attributes = [];

https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/IndexService/Config/AbstractConfig.php#L38

Reply:
yes, I would try making it non-nullable

Reply:
Done